src/operations.py

```diff
@@ -239,10 +239,14 @@ def shift(item, xdata, ydata, left_scale, right_scale, items, ranges):
         previous_item = data_list[index - 1]
 
         # Only use selected span when obtaining values to determine shift value
-        new_xdata, new_ydata = filter_data(
-            previous_item.xdata, previous_item.ydata, ">=", min(xdata))
-        new_xdata, new_ydata = filter_data(
-            new_xdata, new_ydata, "<=", max(xdata))
+        new_xdata, new_ydata = xdata, ydata
```

Comment:
This should perhaps be a seperate PR, but this put is just a bug fix. Basically currently the shift doesn't work properly if a selection is used, and one of the curves is falls outside of the selected range while the other does not. This is due to this filter. So this filter is skipped now if that is the case. That fixes that issue